### ISSUE TITLE
Many Conformance Tests Fail at Once due to Unreachable Node - Tests Time Out After 4h30m

### ISSUE THREAD

**Issue by stevekuznetsov** [openshift/origin#9107]
We don't seem to have an issue for this yet, so I'm making this one to track it so people experiencing this issue on their test runs know that this is a known flake.

There are many runs that exhibit this flake, for example [test_pull_requests_origin_conformance:1459](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/1459/).

We see all extended tests started after some point in time fail, with the OpenShift log showing the following message repeatedly from the point of failure until the point at which our AWS machinery kills the job at 4h 30m:

```
I0526 16:50:45.761156   24459 nodecontroller.go:734] node 172.18.0.248 hasn't been updated for 45.136964512s. Last ready condition is: {Type:Ready Status:Unknown LastHeartbeatTime:2016-05-26 16:49:55 -0400 EDT LastTransitionTime:2016-05-26 16:50:40 -0400 EDT Reason:NodeStatusUnknown Message:Kubelet stopped posting node status.}
```

This always is accompanied by a Docker daemon failure -- the daemon hangs indefinitely and, if you `ssh` into the machine to use the CLI, it won't respond even to `$ docker version`. 

Attaching `strace` to the daemon seems to not be useful, as `strace` wedges at the same time that Docker does. From the Docker daemon stacktraces, we can see that the daemon is hung waiting to respond to client calls:

```
goroutine 0 [idle]:
runtime.futex(0x1ff5638, 0x0, 0x0, 0x0, 0x0, 0x1ff46c0, 0x1, 0x4ad96f, 0x4adb8e, 0x1ff5638, ...)
/usr/lib/golang/src/runtime/sys_linux_amd64.s:277 +0x21 fp=0x7ffe5c7ae3a0 sp=0x7ffe5c7ae398
runtime.futexsleep(0x1ff5638, 0x0, 0xffffffffffffffff)
/usr/lib/golang/src/runtime/os_linux.c:49 +0x47 fp=0x7ffe5c7ae3e8 sp=0x7ffe5c7ae3a0
runtime.notesleep(0x1ff5638)
/usr/lib/golang/src/runtime/lock_futex.go:145 +0xae fp=0x7ffe5c7ae410 sp=0x7ffe5c7ae3e8
stopm()
/usr/lib/golang/src/runtime/proc.c:1178 +0x119 fp=0x7ffe5c7ae430 sp=0x7ffe5c7ae410
findrunnable(0xc208012000)
/usr/lib/golang/src/runtime/proc.c:1487 +0x562 fp=0x7ffe5c7ae468 sp=0x7ffe5c7ae430
schedule()
/usr/lib/golang/src/runtime/proc.c:1575 +0x151 fp=0x7ffe5c7ae498 sp=0x7ffe5c7ae468
goexit0(0xc209918ea0)
/usr/lib/golang/src/runtime/proc.c:1717 +0x16e fp=0x7ffe5c7ae4b0 sp=0x7ffe5c7ae498
runtime.mcall(0x4d3b24)
/usr/lib/golang/src/runtime/asm_amd64.s:186 +0x5a fp=0x7ffe5c7ae4c0 sp=0x7ffe5c7ae4b0

goroutine 18 [IO wait, 71 minutes]:
runtime.gopark(0x4b3310, 0x7ff575acdbd8, 0x156d9b0, 0x7)
/usr/lib/golang/src/runtime/proc.go:130 +0x105 fp=0xc20894f560 sp=0xc20894f530
runtime.netpollblock(0x7ff575acdbb0, 0x72, 0xc20892af88)
/usr/lib/golang/src/runtime/netpoll.go:347 +0x170 fp=0xc20894f5a0 sp=0xc20894f560
runtime.netpollWait(0x7ff575acdbb0, 0x72, 0x8)
/usr/lib/golang/src/runtime/netpoll.go:150 +0x68 fp=0xc20894f5c0 sp=0xc20894f5a0
net.(*pollDesc).Wait(0xc2088edfe0, 0x72, 0x0, 0x0)
/usr/lib/golang/src/net/fd_poll_runtime.go:84 +0x47 fp=0xc20894f5e0 sp=0xc20894f5c0
net.(*pollDesc).WaitRead(0xc2088edfe0, 0x0, 0x0)
/usr/lib/golang/src/net/fd_poll_runtime.go:89 +0x43 fp=0xc20894f608 sp=0xc20894f5e0
net.(*netFD).readMsg(0xc2088edf80, 0xc20892af70, 0x10, 0x10, 0xc20893f620, 0x1000, 0x1000, 0xffffffffffffffff, 0x0, 0x0, ...)
/usr/lib/golang/src/net/fd_unix.go:296 +0x54e fp=0xc20894f6e8 sp=0xc20894f608
net.(*UnixConn).ReadMsgUnix(0xc208043330, 0xc20892af70, 0x10, 0x10, 0xc20893f620, 0x1000, 0x1000, 0x51, 0xc20892ad24, 0x4, ...)
/usr/lib/golang/src/net/unixsock_posix.go:147 +0x167 fp=0xc20894f7a8 sp=0xc20894f6e8
github.com/godbus/dbus.(*oobReader).Read(0xc20893f600, 0xc20892af70, 0x10, 0x10, 0xc20893f600, 0x0, 0x0)
/builddir/build/BUILD/docker-86bbf842e425c6567c726912852976ccfa947e75/vendor/src/github.com/godbus/dbus/transport_unix.go:21 +0xc5 fp=0xc20894f888 sp=0xc20894f7a8
io.ReadAtLeast(0x7ff575a50ec8, 0xc20893f600, 0xc20892af70, 0x10, 0x10, 0x10, 0x0, 0x0, 0x0)
/usr/lib/golang/src/io/io.go:298 +0xf1 fp=0xc20894f8e0 sp=0xc20894f888
io.ReadFull(0x7ff575a50ec8, 0xc20893f600, 0xc20892af70, 0x10, 0x10, 0x51, 0x0, 0x0)
/usr/lib/golang/src/io/io.go:316 +0x6d fp=0xc20894f930 sp=0xc20894f8e0
github.com/godbus/dbus.(*unixTransport).ReadMessage(0xc20892a6d0, 0xc208925080, 0x0, 0x0)
/builddir/build/BUILD/docker-86bbf842e425c6567c726912852976ccfa947e75/vendor/src/github.com/godbus/dbus/transport_unix.go:85 +0x1bf fp=0xc20894fd10 sp=0xc20894f930
github.com/godbus/dbus.(*Conn).inWorker(0xc2089290e0)
/builddir/build/BUILD/docker-86bbf842e425c6567c726912852976ccfa947e75/vendor/src/github.com/godbus/dbus/conn.go:248 +0x58 fp=0xc20894ffd8 sp=0xc20894fd10
runtime.goexit()
/usr/lib/golang/src/runtime/asm_amd64.s:2232 +0x1 fp=0xc20894ffe0 sp=0xc20894ffd8
created by github.com/godbus/dbus.(*Conn).Auth
/builddir/build/BUILD/docker-86bbf842e425c6567c726912852976ccfa947e75/vendor/src/github.com/godbus/dbus/auth.go:118 +0xe84

goroutine 74 [chan receive, 71 minutes]:
runtime.gopark(0x4d56d0, 0xc208a1d0d8, 0x15cdf70, 0xc)
/usr/lib/golang/src/runtime/proc.go:130 +0x105 fp=0xc208a65da8 sp=0xc208a65d78
runtime.goparkunlock(0xc208a1d0d8, 0x15cdf70, 0xc)
/usr/lib/golang/src/runtime/proc.go:136 +0x48 fp=0xc208a65dd0 sp=0xc208a65da8
runtime.chanrecv(0x10bd7e0, 0xc208a1d080, 0xc208a65f18, 0x57fb01, 0x0)
/usr/lib/golang/src/runtime/chan.go:467 +0x82c fp=0xc208a65e70 sp=0xc208a65dd0
runtime.chanrecv1(0x10bd7e0, 0xc208a1d080, 0xc208a65f18)
/usr/lib/golang/src/runtime/chan.go:311 +0x2b fp=0xc208a65ea0 sp=0xc208a65e70
github.com/docker/docker/api/server.(*Server).serveAPI(0xc208068840, 0x0, 0x0)
/builddir/build/BUILD/docker-86bbf842e425c6567c726912852976ccfa947e75/_build/src/github.com/docker/docker/api/server/server.go:108 +0x223 fp=0xc208a65f48 sp=0xc208a65ea0
github.com/docker/docker/api/server.(*Server).Wait(0xc208068840, 0xc208846c60)
/builddir/build/BUILD/docker-86bbf842e425c6567c726912852976ccfa947e75/_build/src/github.com/docker/docker/api/server/server.go:223 +0x2e fp=0xc208a65fd0 sp=0xc208a65f48
runtime.goexit()
/usr/lib/golang/src/runtime/asm_amd64.s:2232 +0x1 fp=0xc208a65fd8 sp=0xc208a65fd0
created by main.(*DaemonCli).CmdDaemon
/builddir/build/BUILD/docker-86bbf842e425c6567c726912852976ccfa947e75/docker/daemon.go:309 +0x2976
```

The `strace` of the daemon hangs during what looks like normal execution [NOTE: since `strace` wedges as well, we do not get complete output from `strace` on the last syscall:

```
12335 12:49:49.953238 execve("/pod", ["/pod"], ["PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "HOSTNAME=docker-registry-1-deploy", "DOCKER_REGISTRY_PORT=tcp://172.30.125.4:5000", "DOCKER_REGISTRY_PORT_5000_TCP=tcp://172.30.125.4:5000", "DOCKER_REGISTRY_PORT_5000_TCP_PORT=5000", "KUBERNETES_SERVICE_PORT_DNS=53", "KUBERNETES_PORT_443_TCP_PORT=443", "KUBERNETES_PORT_53_UDP_PORT=53", "KUBERNETES_PORT_53_TCP_PORT=53", "DOCKER_REGISTRY_SERVICE_HOST=172.30.125.4", "KUBERNETES_PORT_53_TCP_ADDR=172.30.0.1", "KUBERNETES_SERVICE_HOST=172.30.0.1", "KUBERNETES_SERVICE_PORT=443", "KUBERNETES_PORT=tcp://172.30.0.1:443", "KUBERNETES_PORT_443_TCP=tcp://172.30.0.1:443", "KUBERNETES_PORT_443_TCP_ADDR=172.30.0.1", "KUBERNETES_PORT_53_UDP=udp://172.30.0.1:53", "KUBERNETES_PORT_53_UDP_PROTO=udp", "DOCKER_REGISTRY_SERVICE_PORT_5000_TCP=5000", "KUBERNETES_PORT_53_TCP=tcp://172.30.0.1:53", "KUBERNETES_PORT_53_TCP_PROTO=tcp", "KUBERNETES_PORT_53_UDP_ADDR=172.30.0.1", "DOCKER_REGISTRY_PORT_5000_TCP_ADDR=172.30.125.4", "KUBERNETES_SERVICE_PORT_HTTPS=443", "KUBERNETES_PORT_443_TCP_PROTO=tcp", "DOCKER_REGISTRY_PORT_5000_TCP_PROTO=tcp", "KUBERNETES_SERVICE_PORT_DNS_TCP=53", "DOCKER_REGISTRY_SERVICE_PORT=5000", "HOME=/"] <unfinished ...>
7435  12:49:49.953290 <... read resumed> "", 512) = 0
12337 12:49:49.953931 <... select resumed> ) = 0 (Timeout)
7438  12:49:49.953953 <... select resumed> ) = 0 (Timeout)
7435  12:49:49.953964 close(19 <unfinished ...>
12337 12:49:49.953980 clock_gettime(CLOCK_MONOTONIC,  <unfinished ...>
7438  12:49:49.953990 clock_gettime(CLOCK_MONOTONIC,
```

Full versions of the following documents are available:
 - messages to `journald` from the kernel [here](https://drive.google.com/open?id=0B796GVFxYkfANlgwb3NiRHdzZlU)
 - `strace` output for the daemon [here](https://drive.google.com/open?id=0B796GVFxYkfATjFuSmk2LWR3R1k) [WARNING: file is over 1GB]
 - OpenShift log [here](https://drive.google.com/open?id=0B796GVFxYkfAckFLRHNkS1RIUU0)
 - messages to `journald` from the daemon [here](https://drive.google.com/open?id=0B796GVFxYkfASFBiQUFJTXNYOVU) 
 - Docker daemon stacktrace dumped while the process was wedged [here](https://drive.google.com/open?id=0B796GVFxYkfARWRBUlVRUUM3NkU)

**Comment by timothysc** [comment]
cross-link https://bugzilla.redhat.com/show_bug.cgi?id=1341906

**Comment by stevekuznetsov** [comment]
Looks like the issues with the kernel are due to the `strace` attach and not due to the Docker daemon hang by itself.

**Comment by stevekuznetsov** [comment]
Should be fixed with @tdawson's update to Docker 1.10.3-40
